source/faq.txt

```diff
@@ -77,15 +77,19 @@ the driver opens more sockets until the number reaches the minimum.
 You can set the maximum number of milliseconds that a connection can
 remain idle in the pool by using the ``MaxConnectionIdleTime`` option.
 Once a connection is idle for ``MaxConnectionIdleTime``, the driver
-removes and replaces it. This option defaults to 10 minutes.
+removes it. This option defaults to 10 minutes. If the pool size falls
+below ``MinConnectionPoolSize``, the driver removes *and* replaces the
+idle connection.
+
 ``MongoClient`` also has the ``MaxConnectionLifeTime`` option, which
 specifies the length of time, 30 minutes by default, that a connection
 can be pooled before expiring.
 
-A thread that waits more than the length of time defined by
-``MaxConnectionIdleTime`` for a socket raises a connection error. Use this
-option if it is more important to bound the duration of operations
-during a load spike than it is to complete every operation.
+A thread that waits for a socket for the length of time defined by
+the ``ServerSelectionTimeout`` option raises a connection error.
+``ServerSelectionTimeout`` defaults to 30 seconds. Use this option if it
+is more important to bound the duration of operations during a load
+spike than it is to complete every operation.
```

Comment:
`ServerSelectionTimeout` will bound how long an operation will wait to find a matching server, but will not bound the total operation time. If you want to bound the total operation time, you should use `maxTimeMS` (exposed on most operations) and/or `socketTimeoutMS` (not recommended). `maxTimeMS` is exposed as `MaxTime` in the various `FindOptions`, `AggregationOptions`, etc. `socketTimeoutMS` is set via the URI or `MongoClientSettings.SocketTimeout`.